src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -49,6 +55,11 @@ private void StartRaisingEvents()
                 _enabled = true;
 
                 var instance = new RunningInstance(this, _directory, _includeSubdirectories, TranslateFlags(_notifyFilters));
+                if (_cachedRunningInstance.IsAllocated)
+                {
+                    throw new Exception("Should've called StopRaisingEvents before restarting");
```

Comment:
Obviously need a resource string here. But before I add one, do we want to throw here? Should we simply reuse the existing running instance if the user accidentally calls StartRaisingEvents more than once in a row?

Reply:
To today if I call Start twice in a row, do I experience the same behavior as if I called Start once? Ie, it's effectively a no-op? If so then it seems to me that we don't have any good reason to break them. We essentially document that any redundant Start calls are ignored, and preserve that behavior.
If instead today the user gets into a bad state, then starting to throw might be helpful.

Reply:
> Should we simply reuse the existing running instance if the user accidentally calls StartRaisingEvents more than once in a row?

Can you elaborate on what someone's code would look like to call StartRaisingEvents twice in a row?  The method is private.  What path through the code would enable that, other than erroneously using an FSW instance in parallel?